string str
int64 com
